lib/src/components/NxTable/__tests__/NxTable.test.tsx

```diff
@@ -46,6 +46,23 @@ describe('NxTable', function() {
     }
   });
 
+  describe('caption prop', function() {
+    it('renders a <caption> immediately after <table> when caption prop is provided', function() {
+      const component = renderEl({ caption: 'table-title' })!;
+      expect(component.firstElementChild?.tagName).toBe('CAPTION');
+    });
+
+    it('assigns the accessible-name of the table when caption prop is provided', function() {
+      render(<NxTable caption="table-title"></NxTable>);
+      expect(screen.getByRole('table', { name: 'table-title'})).toBeInTheDocument();
+    });
+
+    it('does not render a <caption> when caption prop is not provided', function() {
+      render(<NxTable></NxTable>);
+      expect(document.querySelectorAll('caption').length).toBe(0);
```

Comment:
To match how we use querySelector in other RTL tests, we could do:
```
const {container} = quickRender();
expect(container.querySelector(‘caption’)).not.toBeInTheDocument();
```

Reply:
> To match how we use querySelector in other RTL tests, we could do:
> 
> ```
> const {container} = quickRender();
> expect(container.querySelector(‘caption’)).not.toBeInTheDocument();
> ```

Changed here: 175e2b42d7514aa7d377717ea61d9ab493a8ed02